src/index.rs

```diff
@@ -38,7 +39,7 @@ impl Index {
         Self { inner: index }
     }
 
-    pub fn create_field(&self, name: &str) -> Field {
+    pub fn create_field(&self, name: &str, weight: f64, tag_seperator: Option<i8>, tag_case_sensitive: bool) -> Field {
```

Comment:
A few things here:
1. `tag_seperator` has a typo, should be `tag_separator`. Same for `sperator` below.
2. It should not be an `Option<i8>` but an `Option<char>` (and the `char` should be converted inside).
3. Even better would be to bundle `tag_separator` and `tag_case_sensitive` into a struct describing the tag.